src/mds/OpenFileTable.cc

```diff
@@ -713,12 +717,31 @@ void OpenFileTable::_load_finish(int op_r, int header_r, int values_r,
   try {
     if (first) {
       auto p = header_bl.cbegin();
+
+      string magic;
       version_t version;
       unsigned num_objs;
       __u8 jstate;
-      decode(version, p);
-      decode(num_objs, p);
-      decode(jstate, p);
+
+      if (header_bl.length() == 13) {
+	// obsolete format.
+	decode(version, p);
+	decode(num_objs, p);
+	decode(jstate, p);
+      } else {
+	decode(magic, p);
+	if (magic != CEPH_FS_ONDISK_MAGIC) {
+	  std::ostringstream oss;
+	  oss << "invalid magic '" << magic << "'";
+	  throw buffer::malformed_input(oss.str());
```

Comment:
Right thing to do would be to call `mds->damaged()`?

Reply:
malformed_input will cause mds to re-create openfiletable